sklearn/linear_model/_ridge.py

```diff
@@ -185,11 +186,11 @@ def _solve_cholesky_kernel(K, y, alpha, sample_weight=None, copy=False):
             #       use the fall-back solution below in case a LinAlgError
             #       is raised
             dual_coef = linalg.solve(K, y, sym_pos=True,
-                                     overwrite_a=False)
+                                     overwrite_a=False, check_finite=False)
```

Comment:
`K` itself may not be finite because it it is a dot product:

https://github.com/scikit-learn/scikit-learn/blob/24d58668da03239cb1a45a1bd713d5485478a675/sklearn/linear_model/_ridge.py#L689